extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/trace/ExtendedBaggageBuilder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.incubator.trace;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.util.Map;
+
+public class ExtendedBaggageBuilder {
+  private final BaggageBuilder delegate;
+
+  private ExtendedBaggageBuilder(BaggageBuilder delegate) {
+    this.delegate = delegate;
+  }
+
+  public static ExtendedBaggageBuilder create(BaggageBuilder builder) {
+    return new ExtendedBaggageBuilder(builder);
+  }
+
+  public static ExtendedBaggageBuilder current() {
+    return create(Baggage.current().toBuilder());
+  }
+
+  public ExtendedBaggageBuilder set(String key, String value) {
+    delegate.put(key, value);
+    return this;
+  }
+
+  public ExtendedBaggageBuilder setAll(Map<String, String> baggage) {
```

Comment:
I think this ends up being the only helpful thing we get. I'm comfortable extending `BaggageBuilder` with a `putAll(Map<String, String>)` method an skipping incubation.